### ISSUE TITLE
Code Cleanup Opportunity - UITextFieldDelegate

### ISSUE THREAD

**Issue by jhudsonWA** [WSDOT/HATS#1223]
A ton of View Controllers in the application duplicate the same code. Some do more checks or some do less, but all of them have a similar pattern.

Brian: For your first swift task can you go through all the VCs looking for the following method and delegate.

```
UITextFieldDelegate
    
    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```

Move the logic from one that covers most special UITextFields into a Utility.swift method takes the text field, checks for the particular type and returns true or false based off the duplicated logic. 

EditFeatureStateRouteViewPointViewController.swift has a good example but also pull the checks for others like AttributeNumbersOnlyAcceptedUITextField, OrgcodeUITextField, WholeNumbersOnlyAcceptedUITextField. (Search for usages and copy the logic once for each of these from the VC to the utility method).

Then update all View Controllers so that shouldChangeCharactersIn just calls the Utility method for the return instead of duplicating the logic. 

As always ask if you need more details. **Please do this on a UITextFieldDelegate-Cleanup branch.**

**Comment by jhudsonWA** [comment]
@brianblood Done yet? lol ;P

**Comment by jhudsonWA** [comment]
Merged in. Good job!

**Comment by brianblood** [comment]
Looks good to me.  With any luck, nobody will ever know about this.

On Jan 3, 2019, at 12:48 PM, Joshua Hudson <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #1223<https://github.com/WSDOT/HATS/issues/1223>.

**Comment by jhudsonWA** [comment]
Ok deleting the feature branch. I assigned to more quick ones. Go ahead and create a branch and do both on it.

On Jan 3, 2019, at 12:51 PM, brianblood <notifications@github.com<mailto:notifications@github.com>> wrote:

Looks good to me. With any luck, nobody will ever know about this.

On Jan 3, 2019, at 12:48 PM, Joshua Hudson <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


Closed #1223<https://github.com/WSDOT/HATS/issues/1223>.
